### PR TITLE
feat(source-hubspot): adds individual read scopes to LineItems Stream

### PR DESCRIPTION
**Pull request by tcboles**
## What
This allows oauth tokens to include individual read scopes to lineitems when connecting to hubspot instead of the full e-commerce scope.

## How
Adds additional scopes to line item streams

1. `airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py`

## User Impact
There are not any breaking changes in this PR

## Can this PR be safely reverted and rolled back?
 YES 💚

### CODE REVIEW

**Comment by aldogonzalez8** [comment]
@tcboles Dumb question as I've not worked enough with hubsport scopes:

> This allows oauth tokens to include individual read scopes to lineitems when connecting to hubspot instead of the full e-commerce scope.

I see we still have `scopes = {"e-commerce", "crm.objects.line_items.read"}` but you said instead, will not include both?

**Comment by tcboles** [comment]
> copes = {"e-commer

**Comment by aldogonzalez8** [comment]
> > @tcboles Dumb question as I've not worked enough with hubsport scopes:
> > > This allows oauth tokens to include individual read scopes to lineitems when connecting to hubspot instead of the full e-commerce scope.
> > 
> > 
> > I see we still have `scopes = {"e-commerce", "crm.objects.line_items.read"}` but you said instead, will not include both?
> 
> Hi @aldogonzalez8. Thanks for the reply. My intent here is to allow users to require the `e-commerce` scope or the granular `crm.objects.line_items.read` scope. From my understanding of the code, only one of the configured scopes on the stream are required when reading since its looking for an intersection of the defined scopes with the granted scopes. (See: https://github.com/tcboles/airbyte/blob/251626de830b79087da662ce7012b794521a9f8e/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L369).
> 
> Here is the hubspot documentation for scope requirements for the line_items object. https://developers.hubspot.com/beta-docs/reference/api/crm/objects/line-items#get-%2Fcrm%2Fv3%2Fobjects%2Fline_items.

Thanks, I think that makes sense to me, couple of things.

- Format is falling, can you run locally `airbyte-ci format fix all` to fix the code format. [Docs here](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md).
- You also need yo increase the version in hubspot.md, metadata.yaml and pyproject.toml.

**Comment by tcboles** [comment]
> > > @tcboles Dumb question as I've not worked enough with hubsport scopes:
> > > > This allows oauth tokens to include individual read scopes to lineitems when connecting to hubspot instead of the full e-commerce scope.
> > > 
> > > 
> > > I see we still have `scopes = {"e-commerce", "crm.objects.line_items.read"}` but you said instead, will not include both?
> > 
> > 
> > Hi @aldogonzalez8. Thanks for the reply. My intent here is to allow users to require the `e-commerce` scope or the granular `crm.objects.line_items.read` scope. From my understanding of the code, only one of the configured scopes on the stream are required when reading since its looking for an intersection of the defined scopes with the granted scopes. (See: https://github.com/tcboles/airbyte/blob/251626de830b79087da662ce7012b794521a9f8e/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L369).
> > Here is the hubspot documentation for scope requirements for the line_items object. https://developers.hubspot.com/beta-docs/reference/api/crm/objects/line-items#get-%2Fcrm%2Fv3%2Fobjects%2Fline_items.
> 
> Thanks, I think that makes sense to me, couple of things.
> 
> * Format is falling, can you run locally `airbyte-ci format fix all` to fix the code format. [Docs here](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md).
> * You also need yo increase the version in hubspot.md, metadata.yaml and pyproject.toml.

Thanks [aldogonzalez8](https://github.com/aldogonzalez8). That should be taken care of. Let me know if there is anything else you need from me.

**Comment by aldogonzalez8** [comment]
Thanks @tcboles, I believe you are still missing [pyproject.toml](https://github.com/airbytehq/airbyte/blob/5198b620295e551a6aba3b8a090ed34d393d04c5/airbyte-integrations/connectors/source-hubspot/pyproject.toml#L6) version

```.toml
[tool.poetry]
version = "4.4.1"
```

**Comment by tcboles** [comment]
> Thanks @tcboles, I believe you are still missing [pyproject.toml](https://github.com/airbytehq/airbyte/blob/5198b620295e551a6aba3b8a090ed34d393d04c5/airbyte-integrations/connectors/source-hubspot/pyproject.toml#L6) version
> 
> ```toml
> [tool.poetry]
> version = "4.4.1"
> ```

Sorry about that [aldogonzalez8](https://github.com/aldogonzalez8). I just updated. Thanks!

**Review by aldogonzalez8 — APPROVED** [review]
APPROVED, will merge after release date updated.

File: docs/integrations/sources/hubspot.md [hunk 1]
```diff
@@ -338,6 +338,7 @@ The connector is restricted by normal HubSpot [rate limitations](https://legacyd
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                                                          |
 |:--------|:-----------|:---------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 4.4.2 | 2024-12-09 | [48480](https://github.com/airbytehq/airbyte/pull/48480) | Adds individual read scopes to LineItems Stream |
```
[the remark this answers is not in the document]
Comment (tcboles) [reply docs/integrations/sources/hubspot.md]: Thanks. I just updated.

**Comment by tcboles** [comment]
> APPROVED, will merge after release date updated.

Thanks [aldogonzalez8](https://github.com/aldogonzalez8). I just updated the date.

**Review by bazarnov — APPROVED** [review]

**Pull request merged**